### PR TITLE
Remove deprecated usages of isnan or isinf

### PR DESCRIPTION
**Pull request by perrotta**
#### PR description:

According to the static analyzer, as of CMSSW_11_3_X_2021-04-12-2300 only 7 occurrencies of the deprecated `isnan()` or `isinf()` calls remained in CMSSW

This PR finally gets rid of all them

#### PR validation:

It builds

### CODE REVIEW

**Comment by perrotta** [comment]
please test

File: FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc [hunk 1]
```diff
@@ -378,18 +377,17 @@ bool CTPPSFastTrackingProducer::SearchTrack(int i,
...
   //
   // Calculate the reconstructed track parameters
-  theta = sqrt(thx * thx + thy * thy) * urad;
+  double theta = sqrt(thx * thx + thy * thy) * urad;
   xi = eloss / fBeamEnergy;
   double energy = fBeamEnergy * (1. - xi);
   partP = sqrt(energy * energy - PPSTools::ProtonMassSQ);
   t = -2. * (PPSTools::ProtonMassSQ - fBeamEnergy * energy + fBeamMomentum * partP * cos(theta));
-  pt = sqrt(pow(partP * thx * urad, 2) + pow(partP * thy * urad, 2));
+  pt = partP * theta;
```
Comment (perrotta) [inline FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc:385]: @sandrofonseca this lines reproduces what was coded before
Could you please confirm that the approximation `sin(theta) == theta` was really wanted?

**Comment by perrotta** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc (hunk 1 above)
Comment (perrotta) [reply FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc:385]: > @sandrofonseca this lines reproduces what was coded before
> Could you please confirm that the approximation `sin(theta) == theta` was really wanted?

@jan-kaspar @fabferro FYI

Comment (jan-kaspar) [reply FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc:385]: For forward protons the scattering angles are not larger than O(10^-4 rad), thus the approximation is well justified.

Comment (perrotta) [reply FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc:385]: > For forward protons the scattering angles are not larger than O(10^-4 rad), thus the approximation is well justified.

Thank you Jan: I just wanted to get confirmation that I did not inadvertently hit a typo with this update.

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

File: FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc (hunk 1 above)
Comment (jan-kaspar) [reply FastSimulation/CTPPSFastTrackingProducer/plugins/CTPPSFastTrackingProducer.cc:385]: Thank you for double checking!

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- Technical
- Jenkins tests pass

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
@cms-sw/fastsim-l2 could you please check and sign, if you think so?

**Comment by sandrofonseca** [comment]
@perrotta  I have not involved in the PPS project anymore. I think that @mundim  can help you with it.

**Comment by silviodonato** [comment]
kind reminder @cms-sw/fastsim-l2

**Comment by civanch** [comment]
+1

modifications are technical. PPS part of modifications are correct.

**Comment by qliphy** [comment]
+1

**Pull request merged**